### DISCUSSION TITLE
我使用了 xkb option，但每次锁定屏幕时 option 都会被移除，请问如何取消这样的行为？

### DISCUSSION THREAD

**Discussion by no1xsyzy** [linuxdeepin/developer-center#3472] [General | 常规讨论]
因个人操作习惯，我惯于使用 ```setxkbmap -option ctrl:swapcaps```。当我使用这个命令时，CapsLock 和 Ctrl 确实交换了。
但每当我使用 Super+L 锁定时和再次解锁后，它们就不再交换了。这导致我每次解锁后都必须重复执行这个命令。
我找到了具有类似效果的代码 ```gsettings set com.deepin.dde.keyboard layout-options '["ctrl:swapcaps"]'```，但同样立即产生交换，但不能在锁屏后保持交换。
我认为问题出在 `dde-lock` 上，但我粗略检查和搜索相关代码时只能找到设置 xkbLayout 的代码，却不能找到它因为什么设置了 xkbLayout（找到了一个可能相关的 signal，但找不到会产生这个 signal 的代码）。
请帮助我；如有必要，可提供更多信息。

**Comment by BLumia** [comment]
咨询了下 @justforlxz ，应该是有个“锁屏时允许切换键盘布局”的需求，但设置键盘布局的时候漏掉了布局的 options，按我理解这个算是 bug 😂

**Comment by no1xsyzy** [comment]
发现 Discussions 没有反链，补充一下
已发 issue #3478

**Reply by BLumia** [reply]
assign 给测试那边了，可能还没忙过来 😂
